### ISSUE TITLE
Automate changelog process for repos

### ISSUE THREAD

**Issue by mynktl** [openebs/openebs#2990]

**Comment by mynktl** [comment]
[PR](https://github.com/openebs/velero-plugin/pull/89) raised in openebs/velero-plugin for `master` branch.

**Comment by mynktl** [comment]
> [PR](https://github.com/openebs/velero-plugin/pull/87) raised in openebs/velero-plugin for `master` branch.

Above PR is closed and raised a [new PR](https://github.com/openebs/charts/pull/113) in openebs/charts to _gh-pages_.

**Comment by mynktl** [comment]
Script to raise a changelog PR  : https://raw.githubusercontent.com/openebs/charts/gh-pages/scripts/release/create_changelog_pr.sh

Steps to generate changelog:
- Checkout development(usually `master`) branch 
- Run script with tag name and repo name like,
  ```bash
  ./create_changelog_pr.sh -r openebs/openebs v1.11.0
  ```

  Above command will update the root  changelog  file(`CHANGELOG.md`) and `changelogs/released/v1.11.0`.

  In above command if production(-p)/release(-c) branch is mentioned then script will raise a PR.
  Sample command
  ```bash
  ./create_changelog_pr.sh -r openebs/openebs -p master v1.11.0
  ```


Help manual for script:

```
Usage:
    ./create_changelog_pr.sh <options> TAG-NAME

Example:
    ./create_changelog_pr.sh -r github-org/repo-name v1.1.0-RC1

TAG-NAME is required to execute this script.
Repo Name(-r) is required for github PR link. If it is not mentioned then generated PR link will not work.

Supported options are as below. Mandatory arguments to long options are mandatory for short options too:
    -h, --help                              Show this message and exit
    -r, --repo REPO                         Github repo, in org/repo format.
                                            To raise a PR, this option is mandatory.
                                            If mentioned PR will be created in this repo.
                                            Example:
                                              -r openebs/plugin
                                              --repo openebs/plugin
    -p, --production-branch BRANCH          Name of production branch.
                                            Option -r|--repo is required to use this option.
                                            PR for changelog will be created against this mentioned branch
                                            If options -p|--production-branch and -c|--release-branch, both are
                                            mentioned then two PR will be raised.
                                            Example:
                                              -p master
                                              --production-branch master
    -c, --release-branch RELEASEBRANCH      Name of release branch.
                                            Option -r|--repo is required to use this option.
                                            PR for changelog will be created against this mentioned branch
                                            If options -p|--production-branch and -c|--release-branch, both are
                                            mentioned then two PR will be raised.
                                            Example:
                                              -c v1.10.x
                                              --release-branch v1.10.x


This script expects following environment variable
- USERNAME
    This is your github username, configured through git config --global user.name
    This script doesn't print your USERNAME on output

- TOKEN
    This is github token. This script doesn't print your TOKEN on output.
    To create a new token,
    refer https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line

- EMAIL
    This is your email id, configured through git config --global user.email, to sign the commit
    This script doesn't print your EMAIL on output

How this script works?
- Generating changelog
    Script will use files from directory changelogs/unreleased to create a changelog
    It will create a new directory, named tag-name, under directory changelogs/released
    and move all the files from directory changelogs/unreleased to new directory.
    Once all the files are moved to new directory, it will create a temporary file
    in /tmp/ directory and copy the content of all files, under new directory to a
    temporary file.
    If tag is release tag, like v1.1.0, and there are existing directory for rc release,
    like v1.1.0-RC1, then script will copy the contents of all files, under this rc release
    directory, to a temporary file.
    After this, script will add the content of temporary file to root changelog file
    and delete the temporary file

- Creating a branch
    Script creates a new branch named changelog-timestamp, like changelog-20200512164456

- Creating a commit
    Script will create commit with following files using your USERNAME and EMAIL
    - repo-name/CHANGELOG.md (modified)
    - repo-name/changelogs/unreleased/* (deleted)
    - repo-name/changelogs/released/tag-name (new directory)

- Pushing branch to repo
    To push the changes to github, script will add new remote, named upstream,
    using your USERNAME and TOKEN

- Creating a pull request
    Script uses github API to create a pull request, using your USERNAME and TOKEN.
    Refer https://developer.github.com/v3/pulls/#create-a-pull-request for API
```
